src/Sprites/BitCount.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+
+enum class BitCount : uint16_t
+{
+	One = 1,
+	Four = 4,
+	Eight = 8,
+	Sixteen = 16,
+	TwentyFour = 24,
+	ThirtyTwo = 32
+};
```

Comment:
This seems more like a `uint` field than an `enum`. Granted, the actual expected values tend to be somewhat restricted, so I can kind of see why an `enum` was used here.

In terms of bitmap header values, and other image format headers, I think they typically use a `uint`.

Reply:
I thought about both. I liked the enum because it provides some automatic error checking if someone tries to use a bit count of 3 or 99. It also makes it arguably cleaner if you want to use a switch statement.

Reply:
Sort of yes. An enum field could hold arbitrary integer values outside of what are simply declared in the `enum` though. Granted, that probably falls into the realm of undefined behaviour if you store and use an undeclared value.

For `enum class`, you may get a compiler error on the assignment line for an undeclared name (such as an explicit value), unless it is silenced with a cast. But the same is true in the other direction too. You can't convert from a strongly typed enum back to an integer without a cast. Considering the bit depth is often used in calculations, that could prove to be an annoying hassle.

You can get around the need for a cast when converting back to an integer by using a regular enum, but then you also lose the assignment safety, where arbitrary integer values could be assigned to the enum without error.

I'd say enums make more sense when the distinctions are merely symbolic, with the only restriction on the underlying bit pattern being that they are mutually exclusive of each other.

Reply:
Additionally, I'd like to add these values are loaded from a file, and so only known at runtime. The enum assignment checking is a static check, done at compile time. That means defeating the check with an explicit cast, or maintaining a big switch statement to convert numbers to their identical symbolic name form. The explicit cast, or explicit switch statement must be done in both directions with an `enum class`. Additionally, any switch statements must be maintained if the enum is every updated.

It would be more appropriate to use a uint field with an explicit runtime check and throw an exception when setting this field. You'd only need to worry about assignment, not reading the field.